lib/redmine_merge_request_links/patches/issue_query_patch.rb

```diff
@@ -0,0 +1,65 @@
+module RedmineMergeRequestLinks
+  module Patches
+    module IssueQueryPatch
+      def self.included(base)
+
+        def initialize_available_filters_with_merge_requests
+          initialize_available_filters_without_merge_requests
+          if User.current.allowed_to?(:view_associated_merge_requests, project, :global => true)
+            add_available_filter("merge_request",
+              :type => :list_optional, :values => ['open', 'merged', 'closed']
+            )
+          end
+        end
+
+        def available_columns_with_merge_requests
+          return @available_columns if @available_columns
+          @available_columns = available_columns_without_merge_requests
+          if User.current.allowed_to?(:view_associated_merge_requests, project, :global => true)
+            @available_columns << QueryColumn.new(:merge_requests)
+          end
+          @available_columns
+        end
+
+        def issues_with_merge_requests(options={})
+          if has_column?(:merge_requests)
+            (options[:include] ||= []) << :merge_requests
+          end
+          issues_without_merge_requests(options)
+        end
+
+        def sql_for_merge_request_field(field, operator, value, options={})
+          case operator
+          when "*", "!*"
+            "#{(operator == "*" ? "EXISTS" : "NOT EXISTS")} (SELECT 1 FROM issues_merge_requests" +
+              " LEFT OUTER JOIN projects ON projects.id = issues.project_id" +
+              " WHERE issues_merge_requests.issue_id = issues.id" +
+              " AND " + Project.allowed_to_condition(User.current, :view_associated_merge_requests) + ")"
+          when "=", "!"
+            "EXISTS (SELECT 1 FROM issues_merge_requests" +
+              " JOIN merge_requests ON merge_requests.id = issues_merge_requests.merge_request_id" +
+              " LEFT OUTER JOIN projects ON projects.id = issues.project_id" +
```

Comment:
Which DB do you use? For my this causes an `Unknown column 'issues.project_id' in 'on clause'` error in MySQL 5.7.

Reply:
I am testing with postgres. Can you try again? I replaced the hardcoded table names.

Reply:
From what I've tried on the MySQL console, it looks to me like MySQL does not like the reference to the outer query in the inner query's on clause. I haven't thought deeply about it, but I have a feeling it might be possible to move the `allowed_to_condition` to the outer query.

Reply:
I moved it outside of the subselect and it seems to work.

Reply:
Appears to work now for MySQL.